javascript/modules/client/README.md

```diff
@@ -344,59 +466,173 @@ for await (
     proposalCreationParams,
   )
 ) {
-  switch (step.idx) {
-    case DaoDepositSteps.CREATING:
-      console.log(step.txHash); // 0xb1c14a49...
-      break;
-    case DaoDepositSteps.DONE:
-      console.log(step.proposalId); // 0x1234...
-      break;
+  try {
+    switch (step.idx) {
+      case ProposalCreationSteps.CREATING:
+        console.log(step.txHash); // 0xb1c14a49...
+        break;
+      case ProposalCreationSteps.DONE:
+        console.log(step.proposalId); // 0x1234...
+        break;
+    }
+  }
+  catch (err){
+    // ...err
   }
 }
 ```
 
 ### Voting on an ERC20 proposal
 
-- **TODO**
+```ts
+import {
+  ClientErc20,
+  DaoAction,
+  ICreateProposal,
+  VoteOption,
+  ContextPlugin,
+  VoteProposalStep
+} from "@aragon/sdk-client";
+// context
+const contextPlugin = ContextPlugin.fromContext(context, "0x...")
+const client = new ClientErc20(contextPlugin);
+
+// gas estimation
+const estimatedGas = await client.estimation.voteProposal(
+  proposalId, VoteOptions.YES
+);
+console.log(estimatedGas.average); // bigint
+console.log(estimatedGas.max); // bigint
+
+for await (const step of client.methods.voteProposal(proposalId, VoteOptions.YES)) {
+  try {
+    switch (step.key) {
+    case VoteProposalStep.VOTING:
+        console.log(step.txHash); // 0xb1c14a49...
+      break;
+    case VoteProposalStep.DONE:
+        console.log(step.voteId); // 0x12345...
+      break;
+    }
+  } catch (err) {
+    // ...
+  }
+}
+```
 
 ### Loading the list of members (ERC20)
 
 Retrieving all the members of an ERC20 DAO.
 
 ```ts
-import { ClientErc20 } from "@aragon/sdk-client";
-
-const client = new ClientERC20(context);
-const doaAddressOrEns = "0x1234..."; // unique identifier; dao ENS domain or address
+import { ClientErc20, ContextPlugin } from "@aragon/sdk-client";
+// create client
+const contextPlugin = ContextPlugin.fromContext(context, "0x...")
+const client = new ClientErc20(contextPlugin);
 
-const members = await client.methods.getMembers(daoAddressOrEns);
+const members = await client.methods.getMembers();
 
 console.log(members); // ["0x3585...", "0x1235...", "0x6785...",]
 ```
 
+### Retrieve a proposal by proposalID (ERC20)
+
+Retrieving the proposals of an ERC20 DAO.
+
+```ts
+import { ClientErc20, ContextPlugin } from "@aragon/sdk-client";
+// create client
+const contextPlugin = ContextPlugin.fromContext(context, "0x...")
+const client = new ClientErc20(contextPlugin);
+
+const proposalId = "0x1234567..._0x0"
```

Comment:
Why two terms in the process ID?

Reply:
The `proposalID`has the following format `{address}_{hex_vote_id}` this was to clarify that this function will expect the full id with both parts.

Reply:
Where is this format defined? The UI can decide to serialize/decode data the way it prefers. But speaking from the SDK perspective, we will initialize the `ClientPlugin` with the plugin address, so this address is redundant and the proposal ID can simply consist of the ID itself.